### PR TITLE
net: lib: nrf_cloud: use custom shadow topics

### PR DESCRIPTION
**Pull request by jayteemo**
Use nRF Cloud's new custom shadow topics instead of the generic $aws topics.
IRIS-8718

### CODE REVIEW

**Comment by jayteemo** [comment]
The required cloud changes will be available on Wednesday

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_transport.c [hunk 1]
```diff
@@ -177,13 +149,14 @@ static uint16_t get_message_id(const uint16_t requested_id)
  */
 static void dc_endpoint_free(void)
 {
-	nrf_cloud_free((void *)nct.dc_rx_endp.utf8);
-	nrf_cloud_free((void *)nct.dc_tx_endp.utf8);
-	nrf_cloud_free((void *)nct.dc_m_endp.utf8);
-	nrf_cloud_free((void *)nct.dc_bulk_endp.utf8);
-	nrf_cloud_free((void *)nct.dc_bin_endp.utf8);
+	nrf_cloud_free((void *)nct.dc_eps.base.utf8);
```
Comment (plskeggs) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud_transport.c:167]: you could do this in a `for` loop

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_transport.c [hunk 2]
```diff
@@ -337,41 +310,47 @@ void nct_set_topic_prefix(const char *topic_prefix)
...
-	rejected_topic = NULL;
-	nrf_cloud_free(update_delta_topic);
-	update_delta_topic = NULL;
-	nrf_cloud_free(update_topic);
-	update_topic = NULL;
-	nrf_cloud_free(shadow_get_topic);
-	shadow_get_topic = NULL;
-
+	/* Reset the topics */
+	nrf_cloud_free((void *)nct.cc_eps.accepted.utf8);
```
Comment (plskeggs) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud_transport.c:361]: `for` loop?

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_transport.c [hunk 3]
```diff
@@ -382,22 +361,19 @@ static void nct_topic_lists_populate(void)
 	for (int idx = 0; idx < CC_RX_LIST_CNT; ++idx) {
 		if (nct_cc_rx_opcode_map[idx] == NCT_CC_OPCODE_UPDATE_ACCEPTED) {
```
Comment (plskeggs) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud_transport.c:377]: this would be better as a `switch` statement with 3 cases

**Review by wiba-nordic — APPROVED** [review]

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_transport.c [hunk 4]
```diff
@@ -43,21 +43,36 @@ BUILD_ASSERT((sizeof(CONFIG_NRF_CLOUD_CLIENT_ID) - 1) <= NRF_CLOUD_CLIENT_ID_MAX
 #define NRF_CLOUD_AF_FAMILY AF_INET
 #endif /* defined(CONFIG_NRF_CLOUD_IPV6) */
 
-#define AWS "$aws/things/"
-/*
- * Note that this topic is intentionally not using the AWS Shadow get/accepted
- * topic ("$aws/things/<deviceId>/shadow/get/accepted").
- * Messages on the AWS topic contain the entire shadow, including metadata and
```
[the remark this answers is not in the document]
Comment (jayteemo) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud_transport.c:51]: added to the description of `NCT_SHDW_TPC_GET_TRIM`

**Review by glarsennordic — APPROVED** [review]

**Review by plskeggs — APPROVED** [review]

**Review by tony-le-24 — APPROVED** [review]

**Pull request merged**